### DISCUSSION TITLE
Can brainpoolP256r1 be used?

### DISCUSSION THREAD

**Discussion by msteffenDev** [panva/jose#327] [Q&A]
Hi, I'm trying to use this library to encrypt a token. It's necessary to use brainpoolP256r1.

When I use "BP-256" as curve-type I get the following error:

TypeError [ERR_INVALID_ARG_VALUE]: The property 'key.crv' must be one of: 'P-256', 'secp256k1', 'P-384', 'P-521'. Received 'BP-256'

Is there a solution for this problem?

I use Node.js v17.1.0 and the current version jose-4.3.6.

Thank You!

**Comment by panva** [comment]
Brainpool curves are not registered for use with JOSE. Once they are or there's atleast a WG draft with an intent to register them i'd consider supporting them in the runtimes they're available.

**Reply by MSchmidt** [reply]
Is there a way to add them ourselves in the meantime? Seems like some companies felt the need to use brainpool curves even while not registered with JOSE. Just ran into the same issue.

**Reply by panva** [reply]
> Is there a way to add them ourselves in the meantime?

There is not. This library will not support identifiers that are not bound to land in the IANA registry.

I will give the "`K-256` and `P-256K` crv identifiers" as examples that were chosen in the past by some implementers and what landed in the registry in the end was `secp256k1`, and it was also decided to only support these EC curves for ECDSA and not ECDH.
